terraform/modules/data_share_service/iam.tf

```diff
@@ -213,9 +213,10 @@ data "aws_iam_policy_document" "ecs_task_manage_acquirer_queues" {
   }
 
   statement {
-    sid = "createNewKeys"
+    sid = "createAndDeleteKeys"
     actions = [
-      "kms:CreateKey"
+      "kms:CreateKey",
+      "kms:ScheduleKeyDeletion"
```

Comment:
I think we only want to allow deletion of the correct keys. Something like this: https://docs.aws.amazon.com/kms/latest/developerguide/conditions-kms.html#conditions-kms-resource-aliases:~:text=The%20following%20example%20IAM%20policy%20statement%20allows%20the%20principal%20to%20enable%20and%20disable%20KMS%20keys%20but%20only%20when%20all%20aliases%20of%20the%20KMS%20keys%20include%20%22Test.%22

Reply:
Should have fixed to this